src/astro/utils/table_handler.py

```diff
@@ -16,28 +16,46 @@ def _set_variables_from_first_table(self):
         """
         first_table: Optional[Table] = None
         if self.op_args:
-            table_index = [x for x, t in enumerate(self.op_args) if type(t) == Table]
-            if table_index:
+            table_index = [x for x, t in enumerate(self.op_args) if type(t) is Table]
+            conn_id_set = {x.conn_id for x in self.op_args if type(x) is Table}
+            # Check to see if all tables belong to same conn_id. Otherwise, we this can go wrong for cases
+            # 1. When we have tables from different DBs.
+            # 2. When we have tables from different conn_id, since they can be configured with different
+            # database/schema etc.
+            if table_index and len(conn_id_set) == 1:
                 first_table = self.op_args[table_index[0]]
         elif not first_table:
             table_kwargs = [
                 x
                 for x in inspect.signature(self.python_callable).parameters.values()
                 if (
                     x.annotation == Table
-                    and type(self.op_kwargs[x.name]) == Table
+                    and type(self.op_kwargs[x.name]) is Table
```

Comment:
@utkarsharma2 normally it is better to use `isinstance(self.op_kwargs[x.name], Table)`, because it's faster and also considers inheritance!

Reply:
Sure, make sense.